media_driver/agnostic/gen11/codec/hal/codechal_vdenc_avc_g11.cpp

```diff
@@ -676,7 +676,8 @@ CodechalVdencAvcStateG11::CodechalVdencAvcStateG11(
 
     CODECHAL_ENCODE_ASSERT(m_osInterface);
 
-#if defined(ENABLE_KERNELS) && !defined(_FULL_OPEN_SOURCE)
+#if defined(ENABLE_KERNELS) || defined(_FULL_OPEN_SOURCE)
+    m_freeKernels = true;
```

Comment:
I have rename it to m_enableHmeKernel, and in the base class you can see:
`#if ENABLE_KERNELS
    #if _FULL_OPEN_SOURCE
        m_enableHmeKernel = false;
    #else
        m_enableHmeKernel = true;
    #endif
#endif`

When we only open free kernels, by default m_enableHmeKernel is disabled in the base, because we only enable it for ICL AVC VDEnc, that's why I add the assignment here to enable it for Gen11 AVC VDEnc